### PR TITLE
Add Documents list element

### PR DESCRIPTION
**Pull request by amitaibu**
- Add Documents list paragraph
- `wrapContainerWide`, now accepts an optional background color. If no color then a transparent background is added and no `py` is added, so `verticalSpacing` behaves normally.
- Added `buildElementItemsWithViewMore` as a generic way of adding items that can be hidden with "View more" button

| Style guide | Node view | Node view after "View more" | Node edit |
| -- | -- | -- | -- |
| image | image | image |image |

### CODE REVIEW

**Comment by amitaibu** [comment]
- Added tests.
- Added to migration.

image

**Review by mariano-dagostino — CHANGES_REQUESTED** [review]
@amitaibu Looks good. I finished my review with a few improvements.

File: web/modules/custom/server_general/src/ElementTrait.php [hunk 1]
```diff
@@ -100,6 +100,37 @@ protected function buildElementHeroImage(array $image, string $title, string $su
...
+
+    // Title.
+    $elements[] = $this->buildParagraphTitle($title);
+
+    // Subtitle.
+    $elements[] = $this->wrapTextColor($subtitle, 'dark-gray');
+
+    // Items and "View more" button.
+    $button = $this->buildButton($this->t('View more'), Url::fromUserInput('#'));
+    $elements[] = $this->buildElementItemsWithViewMore($documents, $button, 2);
```
[the remark this answers is not in the document]
Comment (amitaibu) [reply web/modules/custom/server_general/src/ElementTrait.php:127]: I could also remove the view more text, and always use that. My concern was if we'd like to show different kind of button (e.g. primary Vs secondary). But let's be lazy - I'll "hardcode" the button.

Extending projects can change it if needed.

File: web/modules/custom/server_general/src/Plugin/EntityViewBuilder/MediaDocument.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   *   The existing build.
+   * @param \Drupal\media\MediaInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array.
+   */
+  public function buildCard(array $build, MediaInterface $entity): array {
+    /** @var \Drupal\file\FileInterface $file */
+    $file = $this->getReferencedEntityFromField($entity, 'field_media_file');
```
[the remark this answers is not in the document]
Comment (amitaibu) [reply web/modules/custom/server_general/src/Plugin/EntityViewBuilder/MediaDocument.php:37]: I don't think it's needed, PEVB is just reaching to the Media's field.

File: web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php [hunk 3]
```diff
@@ -507,6 +510,33 @@ protected function getHeroImage(): array {
...
.../hEIK8lRjc4Kje1DX5o3s2k8KJCIuYy2FZmZ+D6mi3TDUted15GkBb9OuY2nyj9QUBCCj2/+X2dphspS4dLigoPvwR7MV3HhfunDC77vxPzPTVv3pz+4C2p9Shw0Pv0uKLv/vu3hPvDYi7toROj6IWlNd19eq7mTrrqcOilJhJVUHpK+/c9A/Yc2d+2Imlw6Jpc37wlmAWUzCH9c4VcCBiVh2/kc/a0Sbz5Z0KHKvjZAjoTlp0nnqz3tC7ZnQ/3wDLzjKXeYy2Ag2rUAp111ZIFTAz7vXtuLdJflhUlKMxxdNvvbmvOL67XbGWFQqNxbPP1MN8o/pGcq6ByeVz/Vfu5iGpXWNkAJF96ncwkIObzQZYgTU6Z7189tOzrYnrNR0czhNCYql2Jm0anltuWn2HEa1Vn3iWSu6V40MdnsntnDNEgB3UvTYvxhfiUJVIlRalJxED4wlGDNDCF1jv8yKgkMdYsMzz4/n7PYvj4fwEbbsOX3rBmt6apkdJZVddlK6Msdo4V8bF53zlgncFZ7uzHSK+Dg4BpeWd015Wn5dvLPOX9S5YF71Mzx3lzi4sAhCuN8ATNXM3VUbKFMkmaYah2OA9mMM6KlkmqGsNAwEpC1l9gkNfWpUiig3vPLbtNWyqwMAghCjtLCp7ZsONYT15d2M4Rcmja95+8YdbyXKa138urUMhIlWR+680zqgENiIm2RPzZRplklSU45cKqY1L8wtqpU0WMYGCz5CMa8DMHNyJO/rks55JNmoGUO5404U1P1Izm61t8sHDSo4ofEIbkuU7t1ORGsdJgcCLZUyXjooIEFDrtQjY21VGYlVaiovzx57aVqW+zUXImEUas0lkRFC1Vi18vPnzTgBKtBxX4lu/8tbUmyb0hZC8U4XSJh41f6e1zUi8vfXxp3uJDZ/Eyvf7n+S/+2+bnFhVOLqi0TvJNH7g038cAgTzzOgIa7JDzbMf+t7LB5tb++noBr/w1C+bpfe06fD/AvKxDDLf2mxDvzizWXGqXClTt8Ln93dsNS7LGV1PWKjUlvIQ+aq9cvQwuhLYCtcTFkYagtvWdhWWDoEUIltg7WXPRdt+38nMmpU5gPfI2Tkr8xrh1dESnA3JrMIyJUNE0zzLoisTaZBadi6uwpLDUxtdy3RBQO7U68HXmFdskh68b+3/ATBa1lXVQZ+CAAAAAElFTkSuQmCC',
```
[the remark this answers is not in the document]
Comment (amitaibu) [reply web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:528]: Ok, if that's the reaction - I could load an existing Media image :angel:

**Comment by amitaibu** [comment]
Addressed comments, thanks.

| Node View | Style guide |
| -- | -- |
| image | image|

File: web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php [hunk 4]
```diff
@@ -507,6 +510,51 @@ protected function getHeroImage(): array {
...
+    }
+
+    $media_id = reset($result);
+    /** @var \Drupal\media\MediaInterface $media */
+    $media = $media_storage->load($media_id);
+    $fid = $media->getSource()->getSourceFieldValue($media);
+
+    $file_storage = $this->entityTypeManager()->getStorage('file');
+    /** @var \Drupal\file\FileInterface $file */
+    $file = $file_storage->load($fid);
```
Comment (amitaibu) [inline web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:538]: @mariano-dagostino I think maybe to revert to that scary data URI -- so style guide won't depend on actual content. (Think of this as a `View.elm` -- it shouldn't perform any DB query)

Agree?

Comment (amitaibu) [reply web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php]: https://github.com/Gizra/drupal-starter/pull/445/commits/cc84d21cfefbbaf0a95bbc30c9c6de7b25bae6df

File: web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php [hunk 5]
```diff
@@ -507,6 +510,33 @@ protected function getHeroImage(): array {
...
.../hEIK8lRjc4Kje1DX5o3s2k8KJCIuYy2FZmZ+D6mi3TDUted15GkBb9OuY2nyj9QUBCCj2/+X2dphspS4dLigoPvwR7MV3HhfunDC77vxPzPTVv3pz+4C2p9Shw0Pv0uKLv/vu3hPvDYi7toROj6IWlNd19eq7mTrrqcOilJhJVUHpK+/c9A/Yc2d+2Imlw6Jpc37wlmAWUzCH9c4VcCBiVh2/kc/a0Sbz5Z0KHKvjZAjoTlp0nnqz3tC7ZnQ/3wDLzjKXeYy2Ag2rUAp111ZIFTAz7vXtuLdJflhUlKMxxdNvvbmvOL67XbGWFQqNxbPP1MN8o/pGcq6ByeVz/Vfu5iGpXWNkAJF96ncwkIObzQZYgTU6Z7189tOzrYnrNR0czhNCYql2Jm0anltuWn2HEa1Vn3iWSu6V40MdnsntnDNEgB3UvTYvxhfiUJVIlRalJxED4wlGDNDCF1jv8yKgkMdYsMzz4/n7PYvj4fwEbbsOX3rBmt6apkdJZVddlK6Msdo4V8bF53zlgncFZ7uzHSK+Dg4BpeWd015Wn5dvLPOX9S5YF71Mzx3lzi4sAhCuN8ATNXM3VUbKFMkmaYah2OA9mMM6KlkmqGsNAwEpC1l9gkNfWpUiig3vPLbtNWyqwMAghCjtLCp7ZsONYT15d2M4Rcmja95+8YdbyXKa138urUMhIlWR+680zqgENiIm2RPzZRplklSU45cKqY1L8wtqpU0WMYGCz5CMa8DMHNyJO/rks55JNmoGUO5404U1P1Izm61t8sHDSo4ofEIbkuU7t1ORGsdJgcCLZUyXjooIEFDrtQjY21VGYlVaiovzx57aVqW+zUXImEUas0lkRFC1Vi18vPnzTgBKtBxX4lu/8tbUmyb0hZC8U4XSJh41f6e1zUi8vfXxp3uJDZ/Eyvf7n+S/+2+bnFhVOLqi0TvJNH7g038cAgTzzOgIa7JDzbMf+t7LB5tb++noBr/w1C+bpfe06fD/AvKxDDLf2mxDvzizWXGqXClTt8Ln93dsNS7LGV1PWKjUlvIQ+aq9cvQwuhLYCtcTFkYagtvWdhWWDoEUIltg7WXPRdt+38nMmpU5gPfI2Tkr8xrh1dESnA3JrMIyJUNE0zzLoisTaZBadi6uwpLDUxtdy3RBQO7U68HXmFdskh68b+3/ATBa1lXVQZ+CAAAAAElFTkSuQmCC',
```
Comment (mariano-dagostino) [inline web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:528]:
```suggestion
        Url::fromUserInput('/modules/custom/server_migrate/files/drupal-starter.pdf')->toString(),
```

Comment (mariano-dagostino) [reply web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:528]: @amitaibu :point_up:

Comment (amitaibu) [reply web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:528]: Much nicer!

**Review by mariano-dagostino — APPROVED** [review]

**Pull request merged**